src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -338,22 +335,21 @@ async Task<TimeoutDataEntity> GetTimeoutData(CloudTable timeoutDataTable, string
 
         async Task SaveCurrentTimeoutState(byte[] state, string stateAddress)
         {
-            var container = cloudBlobClient.GetContainerReference(timeoutStateContainerName);
-            var blob = container.GetBlockBlobReference(stateAddress);
+            var blobClient = cloudBlobClient.GetBlobClient(stateAddress);
+            // TODO: Make this more memory efficient
```

Comment:
Memory pool?

Reply:
Well probably RecycleableMemoryStream. But this is a topic for another PR. I just noted a TODO for now. I found quite a few other areas that could benefit an overhaul as well which I haven't noted yet down